### PR TITLE
[SR-10764]  XMLNode: Make the behavior of `name` the same as Darwin.

### PR DESCRIPTION
**Pull request by YOCKOW**
As described in [SR-10764](https://bugs.swift.org/browse/SR-10764), some behavior of `var name: String? { get set }` of `XMLNode` differs between DarwinFoundation and SwiftFoundation.
This PR resolves the issue.

### CODE REVIEW

File: Foundation/XMLDTD.swift [hunk 1]
```diff
@@ -51,7 +51,12 @@ open class XMLDTD : XMLNode {
             //TODO: throw a generic error?
             fatalError("parsing dtd from data failed")
         }
-
+        
+        // _CFXMLParseDTDFromData assigns "none" to DTD's name when there's no name for DTD.
+        if _CFXMLNodeNameEqual(node, "none") {
+            _CFXMLNodeForceSetName(node, nil)
+        }
```
Comment (YOCKOW) [inline Foundation/XMLDTD.swift]: This is a simple way but might not the best way.
If we don't approve it, we have to implement our own DTD parser.

**Comment by spevans** [comment]
@swift-ci test

File: Foundation/XMLDTD.swift (hunk 1 above)
Comment (millenomi) [reply Foundation/XMLDTD.swift]: Nah, it's okay.

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
The test was unnecessary, whoops. I'll merge when done.

**Pull request merged**